app/src/main/java/org/greenstand/android/TreeTracker/database/TreeTrackerDAO.kt

```diff
@@ -136,6 +132,19 @@ interface TreeTrackerDAO {
     @Delete
     fun deleteTreeAttribute(treeAttributeEntity: TreeAttributeEntity)
 
+    @Transaction
+    fun insertTreeWithAttributes(
+        tree: TreeCaptureEntity,
+        attributes: List<TreeAttributeEntity>?
+    ): Long {
+        val treeId = insertTreeCapture(tree)
+        attributes?.forEach {
+            it.treeCaptureId = treeId
+            insertTreeAttribute(it)
+        }
+        return treeId
+    }
+
```

Comment:
This is needed because we want both the inserts succeed together in a transaction. @Transaction I suspect is not supported on any non `Dao` interface or classes. Please correct me if I am wrong.